src/NServiceBus.Core/Pipeline/Outgoing/InspectOutgoingMessageBehavior.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus.Pipeline;
+
+    class InspectOutgoingMessageBehavior : Behavior<IOutgoingPhysicalMessageContext>
```

Comment:
@danielmarbach me and @WojcikMike discusses the perf issues with adding new behaviors but isn't that a flaw of the implementation? 

We felt that setting the header belongs together with https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/Performance/Statistics/ProcessingStatisticsBehavior.cs, perhaps that was wrong and "TimeSent" is some kind of basic "infrastructure" header we put on all messages?

I can see that work if we stop misusing it for audits? (see https://github.com/Particular/NServiceBus/pull/3513#discussion_r54344756)